frontend/src/component/admin/apiToken/ApiTokenPage/ApiTokenPage.tsx

```diff
@@ -1,19 +1,93 @@
-import { useContext } from 'react';
+import { useContext, useMemo } from 'react';
 import AccessContext from 'contexts/AccessContext';
 import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
-import { READ_API_TOKEN } from 'component/providers/AccessProvider/permissions';
+import {
+    CREATE_API_TOKEN,
+    DELETE_API_TOKEN,
+    READ_API_TOKEN,
+} from 'component/providers/AccessProvider/permissions';
 import { AdminAlert } from 'component/common/AdminAlert/AdminAlert';
-import { ApiTokenTable } from 'component/admin/apiToken/ApiTokenTable/ApiTokenTable';
+import { ApiTokenTable } from 'component/common/ApiTokenTable/ApiTokenTable';
+import { PageContent } from 'component/common/PageContent/PageContent';
+import { PageHeader } from 'component/common/PageHeader/PageHeader';
+import { CreateApiTokenButton } from 'component/common/ApiTokenTable/CreateApiTokenButton/CreateApiTokenButton';
+import { Search } from 'component/common/Search/Search';
+import { useApiTokenTable } from 'component/common/ApiTokenTable/useApiTokenTable';
 import { useApiTokens } from 'hooks/api/getters/useApiTokens/useApiTokens';
+import { ActionCell } from 'component/common/Table/cells/ActionCell/ActionCell';
+import { CopyApiTokenButton } from 'component/common/ApiTokenTable/CopyApiTokenButton/CopyApiTokenButton';
+import { RemoveApiTokenButton } from 'component/common/ApiTokenTable/RemoveApiTokenButton/RemoveApiTokenButton';
+import useApiTokensApi from 'hooks/api/actions/useApiTokensApi/useApiTokensApi';
 
 export const ApiTokenPage = () => {
     const { hasAccess } = useContext(AccessContext);
-    const { tokens, loading } = useApiTokens();
+    const { tokens, loading, refetch } = useApiTokens();
+    const { deleteToken } = useApiTokensApi();
```

Comment:
why do we instantiate it here if we later pass it down and never use it ourselves?

Reply:
Because to me this is cleaner than having the RemoveButton keep internal state over which function to call based on which context it's in. Passing in an external onRemove function from the outside allows the component to not care about internal state and have the parent govern the behavior.